tensorboard/plugins/graph/graph_util.py

```diff
@@ -14,152 +14,110 @@
 # ==============================================================================
 """Utilities for graph plugin."""
 
+from tensorboard.compat.proto import graph_pb2
 
-class _ProtoListDuplicateKeyError(Exception):
-    pass
 
+def _prefixed_op_name(prefix, op_name):
+    return "%s/%s" % (prefix, op_name)
 
-class _SameKeyDiffContentError(Exception):
-    pass
 
+def _prefixed_func_name(prefix, func_name):
+    # TODO(stephanwlee): add business logic to strip "__inference_".
+    return "%s_%s" % (prefix, func_name)
 
-def _safe_copy_proto_list_values(dst_proto_list, src_proto_list, get_key):
-    """Safely merge values from `src_proto_list` into `dst_proto_list`.
 
-    Each element in `dst_proto_list` must be mapped by `get_key` to a key
-    value that is unique within that list; likewise for `src_proto_list`.
-    If an element of `src_proto_list` has the same key as an existing
-    element in `dst_proto_list`, then the elements must also be equal.
+def _prepend_names(prefix, orig_graph_def):
+    mut_graph_def = graph_pb2.GraphDef()
+    for node in orig_graph_def.node:
+        new_node = mut_graph_def.node.add()
+        new_node.CopyFrom(node)
+        new_node.name = _prefixed_op_name(prefix, node.name)
+        new_node.input[:] = [
+            _prefixed_op_name(prefix, input_name) for input_name in node.input
+        ]
 
-    Args:
-      dst_proto_list: A `RepeatedCompositeContainer` or
-        `RepeatedScalarContainer` into which values should be copied.
-      src_proto_list: A container holding the same kind of values as in
-        `dst_proto_list` from which values should be copied.
-      get_key: A function that takes an element of `dst_proto_list` or
-        `src_proto_list` and returns a key, such that if two elements have
-        the same key then it is required that they be deep-equal. For
-        instance, if `dst_proto_list` is a list of nodes, then `get_key`
-        might be `lambda node: node.name` to indicate that if two nodes
-        have the same name then they must be the same node. All keys must
-        be hashable.
+        # Remap tf.function method name in the PartitionedCall. 'f' is short for
+        # function.
+        if new_node.op == "PartitionedCall" and new_node.attr["f"]:
+
+            new_node.attr["f"].func.name = _prefixed_func_name(
+                prefix, new_node.attr["f"].func.name,
+            )
+
+    for func in orig_graph_def.library.function:
+        new_func = mut_graph_def.library.function.add()
+        new_func.CopyFrom(func)
+        # Not creating a structure out of factored out function. They already
```

Comment:
Does this comment refer to the fact that we don't use a `/` in the function name prefix, just an underscore?  If so, the comment might make more sense as a comment on `_prefixed_func_name()` itself.